query/src/main/java/org/infinispan/query/clustered/commandworkers/CQDelete.java

```diff
@@ -27,16 +26,14 @@ CompletionStage<QueryResponse> perform(BitSet segments) {
       }
 
       SearchQueryBuilder query = queryDefinition.getSearchQueryBuilder();
-      return blockingManager.supplyBlocking(() -> fetchIds(query), this)
-            .thenApply(queryResult -> queryResult.hits().stream().map(id -> cache.remove(id) != null ? 1 : 0).reduce(0, Integer::sum))
+      return blockingManager.supplyBlocking(() -> fetchIds(query)
+                  .stream().map(id -> cache.remove(id) != null ? 1 : 0).reduce(0, Integer::sum), this)
```

Comment:
```suggestion
.thenCompose(queryResult -> CompletionStages.performSequentially(queryResult.hits().iterator(), cache::removeAsync,
                  Collectors.summingInt(prev -> prev != null ? 1 : 0)))
```
We should be able to keep this non blocking with the following and have the same behavior as before.